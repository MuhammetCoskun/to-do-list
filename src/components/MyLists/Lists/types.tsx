import { ToDoListItemType } from "../../ShowList/ToDoList/types";

export type ListType = {
  id: string;
  title: string;
  todos: ToDoListItemType[];
};

export type ListsProps = {
  lists: ListType[];
  onChooseList: (arg: string) => void;
};
