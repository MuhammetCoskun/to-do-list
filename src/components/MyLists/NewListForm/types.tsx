export type NewListFormProps = {
  onAddList: (listName: string) => void;
};
