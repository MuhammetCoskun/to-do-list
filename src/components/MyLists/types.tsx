import { ListType } from "./Lists/types";

export type MyListsProps = {
  onChooseList: (arg: string) => void;
  onAddList: (title: string) => void;
  lists: ListType[];
};
