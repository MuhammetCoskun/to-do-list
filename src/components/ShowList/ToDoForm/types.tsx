export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
