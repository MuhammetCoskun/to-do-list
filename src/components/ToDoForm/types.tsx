export interface TodoFormProps {
  onAddTodo: (todo: string) => void;
}
