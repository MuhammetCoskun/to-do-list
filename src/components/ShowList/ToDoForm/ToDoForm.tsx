import React, { useRef, useState } from "react";
import { ToDoFormProps } from "./types";

const ToDoForm: React.FC<ToDoFormProps> = ({
  onAddTodo,
  onRemoveTodo,
  onRemoveList,
}) => {
  const todoRef = useRef<HTMLInputElement>(null);
  const [inputIsValid, setInputIsValid] = useState<boolean>(true);

  const submitHandler = (event: React.FormEvent) => {
    event.preventDefault();
    const enteredTodo = todoRef.current!.value;
    if (enteredTodo?.trim().length === 0) {
      setInputIsValid(false);
      return;
    }
    setInputIsValid(true);
    onAddTodo(enteredTodo);
    todoRef.current!.value = "";
  };

  return (
    <div className={"task-form"}>
      <form onSubmit={submitHandler}>
        <input
          className="new todo"
          type="text"
          placeholder="Enter a new to do"
          aria-label="new to do name"
          ref={todoRef}
          data-testid="input"
        />
        <button
          aria-label="create new to do"
          data-testid="btn-create"
          className="btn create"
        >
          +
        </button>
      </form>
      <div className="delete-item-container">
        <button
          className="btn delete"
          onClick={onRemoveTodo}
          data-testid="btn-delete-todo"
        >
          Clear completed tasks!
        </button>
        <button
          className="btn delete"
          onClick={onRemoveList}
          data-testid="btn-delete-list"
        >
          Delete list!
        </button>
      </div>
      {!inputIsValid && (
        <p data-testid="paragraph">Please enter a valid to do..</p>
      )}
    </div>
  );
};

export default ToDoForm;
