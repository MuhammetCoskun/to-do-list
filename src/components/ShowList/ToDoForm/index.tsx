import React, { useRef, useState } from "react";
import { ToDoFormProps } from "./types";

const ToDoForm: React.FC<ToDoFormProps> = ({ onAddTodo }) => {
  const todoRef = useRef<HTMLInputElement>(null);
  const [inputIsValid, setInputIsValid] = useState<boolean>(true);

  const submitHandler = (event: React.FormEvent) => {
    event.preventDefault();
    const enteredTodo = todoRef.current!.value;
    if (enteredTodo?.trim().length === 0) {
      setInputIsValid(false);
      return;
    }
    setInputIsValid(true);
    onAddTodo(enteredTodo);
    todoRef.current!.value = "";
  };

  return (
    <div className={"task-form"}>
      <form onSubmit={submitHandler}>
        <input
          className="new todo"
          type="text"
          placeholder="Enter a new to do"
          aria-label="new to do name"
          ref={todoRef}
          data-testid="input"
        />
        <button
          aria-label="create new to do"
          data-testid="button"
          className="btn create"
        >
          +
        </button>
      </form>
      <div className="delete-item-container">
        <button className="btn delete">Clear completed tasks!</button>
        <button className="btn delete">Delete list!</button>
      </div>
      {!inputIsValid && (
        <p data-testid="paragraph">Please enter a valid to do..</p>
      )}
    </div>
  );
};

export default ToDoForm;
//TODO add style
