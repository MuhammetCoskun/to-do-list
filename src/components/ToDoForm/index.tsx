import React, { useRef, useState } from "react";
import { ToDoFormProps } from "./types";

const ToDoForm: React.FC<ToDoFormProps> = ({ onAddTodo }) => {
  const todoRef = useRef<HTMLInputElement>(null);
  const [inputIsValid, setInputIsValid] = useState<boolean>(true);

  const submitHandler = (event: React.FormEvent) => {
    event.preventDefault();
    const enteredTodo = todoRef.current!.value;
    if (enteredTodo?.trim().length === 0) {
      setInputIsValid(false);
      return;
    }
    setInputIsValid(true);
    onAddTodo(enteredTodo);
    todoRef.current!.value = "";
  };

  return (
    <form onSubmit={submitHandler}>
      <div>
        <input
          type="text"
          placeholder="Enter a to do!"
          ref={todoRef}
          data-testid="input"
        />
      </div>
      <button type="submit" data-testid="button">
        Enter
      </button>
      {!inputIsValid && (
        <p data-testid="paragraph">Please enter a valid to do..</p>
      )}
    </form>
  );
};
//TODO add style and 'data-testid's
export default ToDoForm;
