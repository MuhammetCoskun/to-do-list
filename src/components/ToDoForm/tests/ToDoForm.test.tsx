import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ToDoForm from "../index";
import { TodoFormProps } from "../types";

const mockOnAddTodo = jest.fn();
const customRender = (props: TodoFormProps) => {
  render(<ToDoForm {...props} />);
};
describe("ToDoForm component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<ToDoForm onAddTodo={mockOnAddTodo} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders error message on invalid input", () => {
    customRender({ onAddTodo: mockOnAddTodo });
    const buttonElement = screen.getByRole("button");
    userEvent.click(buttonElement);
    expect(
      screen.getByText("Please enter a valid to do..")
    ).toBeInTheDocument();
  });
  test("when button clicked it should call onAddTodo callback", () => {
    customRender({ onAddTodo: mockOnAddTodo });
    const inputElement = screen.getByTestId("input");
    const buttonElement = screen.getByTestId("button");
    userEvent.type(inputElement, "Write Tests!");
    userEvent.click(buttonElement);
    expect(mockOnAddTodo).toHaveBeenCalledWith("Write Tests!");
  });
});
