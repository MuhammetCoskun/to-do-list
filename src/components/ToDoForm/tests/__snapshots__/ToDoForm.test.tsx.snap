// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoForm component renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div>
      <input
        data-testid="input"
        placeholder="Enter a to do!"
        type="text"
      />
    </div>
    <button
      type="submit"
    >
      Enter
    </button>
  </form>
</DocumentFragment>
`;
